app/code/Magento/Quote/Model/ShippingAddressManagement.php

```diff
@@ -117,10 +117,6 @@ public function assign($cartId, \Magento\Quote\Api\Data\AddressInterface $addres
         $address->setSaveInAddressBook($saveInAddressBook);
         $address->setCollectShippingRates(true);
 
-        if (!$quote->validateMinimumAmount($quote->getIsMultiShipping())) {
```

Comment:
The only other check for Minimum Amount occurs in the Checkout Index controller and it is not sufficient, since checkout flow may be performed using Web API without calling the controller.
Please consider moving this check to Place Order call.